docs/metrics.md

```diff
@@ -0,0 +1,587 @@
+  metrics in any backend system:
+  
+  * **Total Accuracy**: OpenTelemetry Metrics always ensures the total
+    aggregation (sum of metric values across all dimensions) remains accurate,
+    even when overflow occurs.
+  
+  * **Attribute-Based Query Limitations**: Any metric query based on specific
+    attributes could be misleading, as it's possible those dimensions were
+    folded into the overflow bucket due to cardinality capping.
+  
+  * **All Attributes Affected**: When overflow occurs, it's not just
+    high-cardinality attributes that are affected. The entire attribute
+    combination is replaced with the `{"otel.metric.overflow": true}` attribute,
+    meaning queries for any attribute in that combination will miss data points.
+
+#### Cardinality Limits - Example
+
+Extending our fruit sales tracking example, imagine we set a
+cardinality limit of 3 and we're tracking sales with attributes for `name`,
+`color`, and `store_location`:
+
+During a busy sales period at time (T3, T4], we record:
+  
+1. 10 red apples sold at Downtown store
+2. 5 yellow lemons sold at Uptown store
+3. 8 green apples sold at Downtown store
+4. 3 red apples sold at Midtown store (at this point, the cardinality limit is
+    hit, and attributes are replaced with overflow attribute.)
+
+The exported metrics would be:
+  
+* attributes: {name = `apple`, color = `red`, store_location = `Downtown`},
+    count: `10`
+* attributes: {name = `lemon`, color = `yellow`, store_location = `Uptown`},
+    count: `5`
+* attributes: {name = `apple`, color = `green`, store_location = `Downtown`},
+    count: `8`
+* attributes: {`otel.metric.overflow` = `true`}, count: `3`
+  
+  If we later query "How many red apples were sold?" the answer would be 10, not
+  13, because the Midtown sales were folded into the overflow bucket. Similarly,
+  queries about "How many items were sold in Midtown?" would return 0, not 3.
+  However, the total count across all dimensions (i.e How many total fruits were
+  sold in (T3, T4] would correctly give 26) would be accurate.
+
+  This limitation applies regardless of whether the attribute in question is
+  naturally high-cardinality. Even low-cardinality attributes like "color"
+  become unreliable for querying if they were part of attribute combinations
+  that triggered overflow.
+
+  OpenTelemetry's cardinality capping is only applied to attributes provided
+  when reporting measurements via the [Metrics API](#metrics-api). In other
+  words, attributes used to create `Meter` or `Resource` attributes are not
+  subject to this cap.
+
+  :heavy_check_mark: Use `Meter` attributes or `Resource` attributes as
+  appropriate, see [modelling attributes](#modelling-metric-attributes) for
+  details. In the above example, if a process only sells fruits from a
+  particular store, then store_location attribute should be modelled as a
+  Resource - this is not only efficient, but reduced the cardinality capping
```

Comment:
Attributes which can be modeled as `Meter` attributes or `Resource` attributes are static and do not affect cardinality. This should be done primarily for performance (reducing lookup costs). We should have this point, but it should be placed outside of cardinality capping section.